### ISSUE TITLE
Hooking up a MediaSource for specific subtitles when a video does not have them causes looping to fail after the 2nd play through

### ISSUE THREAD

**Issue by Ste-RH** [google/ExoPlayer#7590]
### [REQUIRED] Issue description
Defining a subtitle media source of specific type/language (via createTextSampleFormat) causes a video to not loop a second time.  If I pass NULL into the 'sampleMimeType and 'language' the video will loop perfectly over and over and over again as it should do.

### [REQUIRED] Reproduction steps
I changed the demo code to loop and added the following video to media.exolist.json:
`
      {
        "name": "test",
        "uri": "https://html5demos.com/assets/dizzy.mp4",
        "subtitle_uri": "https://html5demos.com/assets/dizzy.mp4",
        "subtitle_mime_type": "text/vtt",
        "subtitle_language": "en"
      }
`

### [REQUIRED] Link to test content
Using: https://html5demos.com/assets/dizzy.mp4

### [REQUIRED] A full bug report captured from the device
[testreport.txt](https://github.com/google/ExoPlayer/files/4869523/testreport.txt)  The important part of the log can be seen by searching for 'TextRenderer' I think.

### [REQUIRED] Version of ExoPlayer being used
2.11.5

### [REQUIRED] Device(s) and version(s) of Android being used
Nexus 7 (2nd gen), Android Version: 6.0.1

**Comment by icbaker** [comment]
I can't reproduce the problem. I added `player.setRepeatMode(Player.REPEAT_MODE_ALL);` here:
https://github.com/google/ExoPlayer/blob/6026b919a143ad870cfad3a9d84a260452ebc43b/demos/main/src/main/java/com/google/android/exoplayer2/demo/PlayerActivity.java#L386

I tried with both `REPEAT_MODE_ALL` and `REPEAT_MODE_ONE`.

I see the video play (with no visible subtitles, as expected). When it reaches the end it plays again.

Aside: I'm not sure passing an MP4 file with embedded WebVTT subtitles to `subtitle_uri` is supported - that option is designed for side-loading of subtitle text files as described here:
https://exoplayer.dev/media-sources.html#side-loading-a-subtitle-file

The error you see in the log seems like ExoPlayer is trying to decode 'random' bytes as a WebVTT file and is upset that it doesn't start with the `WEBVTT` magic string. That may happen if it tries to decode a full MP4 file as a WebVTT file

Would you be able to clarify your reproduction steps to help me understand the problem?

**Comment by Ste-RH** [comment]
It is the end of the second play-through that then does not loop.

So let the mp4 play through once, loop, play again...it's at the end of that loop where it fails to loop again.

I originally came across this issue loading an HLS stream that did not include a subtitle track (I was told it did)...the MP4 shown is  just a quicker (shorter) was of reproducing the same issue.

**Comment by icbaker** [comment]
Ah thanks - I'd misunderstood "loop a second time".

I can now reproduce. I can also reproduce the same behaviour without changing the repeat mode by seeking during the first playback.

I also noticed that at the end of the playback it doesn't fail/crash, the position keeps counting up beyond the end of the video suggesting it thinks there's still subtitle data to decoder.

I worked out it's related to this change (before this playback would fail immediately at the start of the first playback):
https://github.com/google/ExoPlayer/commit/4107375c9d21136090d0071b04a498024eec6a4c

And specifically the fact that we set `SimpleDecoder#exception = null` in `SimpleDecoder#flush()` but if we've seen an error then the `decoderThread` has already exited, so all future calls to `SimpleDecoder#queueInputBuffer()` (and most other methods) don't throw an exception but also don't trigger any concrete decoding work. This means we never receive an end-of-stream signal, so the TextRenderer keeps playback alive (which is why the counter keeps counting up at the end).

I'll work on fixing this - thanks for reporting.
